### ISSUE TITLE
MongoDB tracing doesn't work if the client has been configured with a CommandListener

### ISSUE THREAD

**Issue by krosen040** [DataDog/dd-trace-java#3361]
Our MongoDB instrumentation stopped working after upgrading from 0.88.0 to 0.92.0, I'm guessing due to https://github.com/DataDog/dd-trace-java/pull/3110.

I believe [this](https://github.com/DataDog/dd-trace-java/blob/v0.94.1/dd-java-agent/instrumentation/mongo/common/src/main/java/datadog/trace/instrumentation/mongo/MongoCommandListener.java#L91-L110) is the problematic code. There appears to be several problems here:

* I don't think the `while` will ever be entered, since `hasPrevious` will be `false` when on the first element
* Even if the `while` is entered, the instrumented `CommandListener` will only be added if there is another `datadog.trace.instrumentation.mongo.MongoCommandListener`

So in conclusion, this only works correctly where there's no other `CommandListener` configured.

**Comment by charliegracie** [comment]
I will look into fixing this issue.

**Comment by charliegracie** [comment]
The fix has been merged and will be available in the next release
